supabase/migrations/20240530291033_add_completed_parcels_view.sql

```diff
@@ -0,0 +1,30 @@
+CREATE OR REPLACE VIEW completed_parcels AS
+WITH completed_events AS (
+    SELECT
+        events.parcel_id,
+        MAX(events.timestamp) AS completed_timestamp
+    FROM
+        events
+    WHERE
+        new_parcel_status IN (
+            SELECT
+                event_name
+            FROM
+                status_order
+            WHERE
+                is_successfully_completed_event
+        )
+    GROUP BY
+        events.parcel_id
+)
+SELECT
+    completed_events.parcel_id,
+    completed_timestamp,
+    family_count,
+    pet_food
+FROM
+    completed_events
+        INNER JOIN
+    parcels_plus ON parcels_plus.parcel_id = completed_events.parcel_id AND NOT parcels_plus.last_status_event_name = 'Parcel Deleted'
+        LEFT JOIN
+    clients ON parcels_plus.client_id = clients.primary_key
```

Comment:
Simplest SQL I can come up with for this is 
```
SELECT events.parcel_id, events.timestamp AS completed_timestamp, family_count, pet_food
FROM events 
INNER JOIN parcels_plus ON events.parcel_id = parcels_plus.parcel_id
LEFT JOIN clients ON parcels_plus.client_id = clients.primary_key
WHERE new_parcel_status LIKE 'Parcel Collected' or new_parcel_status LIKE 'Fulfilled with Trussell Trust' or new_parcel_status like 'Delivered'
```

but I don't know if it works in views and also don't know if it returns useful data at all

Reply:
Will have a test of this :)

Reply:
Have just tested and it does work! Though there's a couple of edge cases that it doesn't cover, but am not sure how much that they matter:
1. Parcels will be double counted if they get set to multiple successful states, although this probably shouldn't happen, there's nothing to stop you in the website from setting delivered twice, delivered then collected. That's why I group_by parcel_id to ensure uniqueness, and set to MAX(timestamp) so the completion time is set to the last time the parcel is set to completed. 
2. Parcels will not be removed if they are deleted (why I added AND NOT parcels_plus.last_status_event_name = 'Parcel Deleted'.

Lmk if any of these edge cases actually matter cause agree that the above is a nice simplification in that case. If we keep the above query - should probably add a comment to explain why I've done it. Thanks!

Reply:
Just noticed another edge case - Deleted Clients. Currently they count in the total but not any of the family sized (as that info has been removed), or any of the pets data (as that info has been removed). Is this how it should work?

Reply:
Cool cool - happy to remove the deleted parcels clause, that makes sense.  Though the flip side of this is that there is now no way of removing parcels from being counted by the reports. Will do that change though.

Re deleted clients, we saying leave it as it is for now? Yeah I see that disadvantage of using a view like I have. One could have mix of live data from the last month and stored data before then, or just a stored table (with a similar sql definition) and a cron job, but I think that sounds like a future ticket? Happy to make those extra tickets to tidy this up once this is merged.

Reply:
Yeah I think we leave it as is (doubt they'll be deleting clients anytime soon) and then create a ticket to aggregate a "permanent" report somewhere